### PR TITLE
Implement the Login UI

### LINKED ISSUE #4: Implement the Login UI
- [x] Login screen

Image

### PR DESCRIPTION
**Pull request by truongtdph40209**
## Related issues
Resolved #4 

## Summary
Implement the Login UI

### Steps:
1. Open `Index.js` and delete the code in render
2. **Paste** the following line:

```jsx
<LoginTemplate/>
```
3. Run app

## Screenshots
| Light mode | Dark mode |
| --- | --- |
|image|image|

### CODE REVIEW

File: components/atoms/InputWithIcons.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    >
+      {!isNil(props.iconRight) && props.iconRight}
+
+      <Input
+        {...props}
+        unstyled
+        marginHorizontal={18}
+        height={54}
+        fontSize={16}
+        color={'#156778'}
```
Comment (KMTus) [inline components/atoms/InputWithIcons.tsx:39]: color ?  Please do not use the code

File: components/molecules/InputForm.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      <IconVisiablePassword
+        size={16}
+        color={colors.primaryTeal}
+        onPress={() => { setShowPassword(!showPassword) }}
+      />
+    )
+  }
+
+  return (
+    <View>
```
Comment (KMTus) [inline components/molecules/InputForm.tsx:41]: Change the view with tamagui

File: constants/Colors.ts [hunk 3]
```diff
@@ -6,12 +6,22 @@ const colors: Record<'light' | 'dark', Colors> = {
   dark: {
     blueSapphire: '#156778',
     gray: '#757575',
-    text: '#fff'
+    lightMist: '#F0F3F6',
+    primaryTeal: '#156778',
+    slateGray: '#50555C',
+    text: '#fff',
```
Comment (KMTus) [inline constants/Colors.ts:12]: text = #fff ?

File: constants/Colors.ts [hunk 4]
```diff
@@ -6,12 +6,22 @@ const colors: Record<'light' | 'dark', Colors> = {
...
+
   },
   light: {
     blueSapphire: '#156778',
     gray: '#757575',
-    text: '#000'
+    lightMist: '#F0F3F6',
+    primaryTeal: '#156778',
+    slateGray: '#50555C',
+    text: '#000',
```
Comment (KMTus) [inline constants/Colors.ts:22]: text = #000 ?

**Comment by KMTus** [comment]
image
Losing the selection without an account, using Scolllview bar to optimize screens

**Comment by KMTus** [comment]
image
UI ok

File: constants/Colors.ts (hunk 3 above)
Comment (truongtdph40209) [reply constants/Colors.ts:12]: yes

File: constants/Colors.ts (hunk 4 above)
Comment (truongtdph40209) [reply constants/Colors.ts:22]: yes

**Review by KMTus — APPROVED** [review]

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
This PR is terrible. The font family is wrong, the UI doesn't match the design, and miscellaneous names are set for the component...

File: components/molecules/InputForm.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+import { Image, Text, View, YStack } from 'tamagui'
+
+import InputWithIcons from '~/components/atoms/InputWithIcons'
+import { NegativeButton } from '~/components/atoms/NegativeButton'
+import { PositiveButton } from '~/components/atoms/PositiveButton'
+import { TextTitle } from '~/components/atoms/TextTitle'
+import getColors from '~/constants/Colors'
+import useTranslation from '~/hooks/useTranslation'
+
+interface Props {
```
Comment (NguyenVanViet2004) [inline components/molecules/InputForm.tsx:14]: Use `type` instead of `interfaces`, it look like 

```suggestion
type Props =  {
 ....
} & InputProps
```

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]

File: locales/vi.json [hunk 6]
```diff
@@ -25,6 +25,23 @@
...
+            "loginPrompt": "Rất vui được gặp lại bạn!, vui lòng đăng nhập để sử dụng ứng dụng.",
+            "email": "Email của bạn",
+            "password": "Mật khẩu của bạn",
+            "forgotPassword": "Quên mật khẩu?",
+            "signIn": "Đăng nhập",
+            "signInWithGoogle": "Đăng nhập bằng google",
+            "signupPrompt": "Bạn chưa có tài khoản?",
+            "joinNow": "Đăng ký ngay"
+        },
+        "signup": {
```
Comment (NguyenVanViet2004) [inline locales/vi.json:40]: Use cameCase

```suggestion
        "signUp": {
```

File: components/molecules/common/Footer.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+import getColors from '~/constants/Colors'
+
+interface Props {
+  title: string
+  subtitle?: string
+  onPressAuthScreen?: () => void
+}
+
+const Footer: React.FC<Props> = (
```
Comment (NguyenVanViet2004) [inline components/molecules/common/Footer.tsx:13]: `Footer` for what??? Rename

File: components/atoms/TextTitle.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import { useColorScheme } from 'react-native'
+import { Text, type TextProps } from 'tamagui'
+
+import getColors from '~/constants/Colors'
+
+type Props = {
+  text: string
+} & TextProps
+
+export const TextTitle = (props: Props): React.ReactElement => {
```
Comment (NguyenVanViet2004) [inline components/atoms/TextTitle.tsx:11]: Did you mean `Header`?

File: components/atoms/InputWithIcons.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    borderWidth: isFocused ? 1 : 0
+  }
+
+  return (
+    <XStack
+      alignItems="center"
+      borderRadius={10}
+      paddingHorizontal={24}
+      style={inputContainerStyle}
+
```
Comment (NguyenVanViet2004) [inline components/atoms/InputWithIcons.tsx:29]: Delete empty space

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]

File: components/atoms/TextTitle.tsx (hunk 8 above)
Comment (truongtdph40209) [reply components/atoms/TextTitle.tsx:11]: no i don't mean `header`

File: package.json
Comment (NguyenVanViet2004) [inline package.json:1]: Why are there 2 new libraries here? Are they necessary? Where to use?

File: interfaces/Colors.ts [hunk 10]
```diff
@@ -9,4 +9,7 @@ export interface Colors {
   lightTransparentBlack: string
   backGround: string
   inputBackground: string
+  black: string
+  textButton: string
```
Comment (NguyenVanViet2004) [inline interfaces/Colors.ts:13]: Rename

File: components/molecules/common/Footer.tsx
Comment (NguyenVanViet2004) [inline components/molecules/common/Footer.tsx:1]: `Footer` for what?

File: components/templates/LoginTemplate.tsx [hunk 11]
```diff
@@ -21,42 +22,48 @@ const LoginTemplate: React.FC = (): JSX.Element => {
...
+            visibleInputWithIcons={false}
+            visibleForgotPassword={true}
+          />
 
-        <AuthButtonsGroup
-          onLoginPress={ButtonLogin}
-          onLoginGooglePress={ButtonLogin}
-        />
-      </View>
+          <AuthButtonsGroup
```
Comment (NguyenVanViet2004) [inline components/templates/LoginTemplate.tsx:45]: Replace `AuthButtonsGroup` with [Separator](https://tamagui.dev/ui/separator)

```suggestion
          <Separator />
```

File: components/molecules/AuthButtonsGroup.tsx
Comment (NguyenVanViet2004) [inline components/molecules/AuthButtonsGroup.tsx:1]: Delete this component, it unnecessary

File: components/atoms/NegativeButton.tsx [hunk 12]
```diff
@@ -24,6 +24,8 @@ export const NegativeButton = (props: Props): React.ReactElement => {
       borderWidth={1}
       borderColor={colors.primaryTeal}
       borderRadius={RADIUS_BUTTON}
+      backgroundColor={colors.white}
```
Comment (NguyenVanViet2004) [inline components/atoms/NegativeButton.tsx:27]: Delete the following two lines of code. After that, you can set the `backgroundColor` and `color` as shown below:

 ```jsx
<NegativeButton backgroundColor="#fff"  color="#fff"/>
```

```suggestion
      backgroundColor={colors.white}
      color={colors.textButton}
```

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]

File: components/molecules/common/Footer.tsx
Comment (truongtdph40209) [reply components/molecules/common/Footer.tsx:1]: I use it as a component that contains `text`

Comment (truongtdph40209) [reply components/molecules/common/Footer.tsx:1]: i will rename it

File: components/atoms/PositiveButton.tsx [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+  title: string
+} & ButtonProps
+
+export const PositiveButton = (props: Props): React.ReactElement => {
+  const colors = getColors(useColorScheme())
+  return (
+    <Button
+      {...props}
+      unstyled
+      pressStyle={{ backgroundColor: colors.gray }}
```
Comment (KMTus) [inline components/atoms/PositiveButton.tsx:18]: similar

File: components/atoms/InputWithIcons.tsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+      style={inputContainerStyle}
+    >
+      {!isNil(props.iconRight) && props.iconRight}
+
+      <Input
+        {...props}
+        unstyled
+        marginHorizontal={18}
+        paddingVertical={14}
+        fontFamily={'JetBrainsMonoRegular'}
```
Comment (KMTus) [inline components/atoms/InputWithIcons.tsx:37]: use hook [`useAppFont()`](https://github.com/NguyenVanViet2004/FE-GlowUp/blob/main/components/templates/HomeTemplate.tsx) for set fontFamily

File: components/atoms/InputWithIcons.tsx [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+  const inputContainerStyle = {
+    backgroundColor: isFocused ? colors.inputBackground : colors.lightMist,
+    borderColor: isFocused ? colors.primaryTeal : colors.lightMist,
+    borderWidth: isFocused ? 1 : 0
+  }
+
+  return (
+    <XStack
+      alignItems="center"
+      borderRadius={10}
```
Comment (thanhdo29) [inline components/atoms/InputWithIcons.tsx:28]: use size in Constants file

**Review by thanhdo29 — APPROVED** [review]

**Review by KMTus — APPROVED** [review]

**Review by thanhdo29 — APPROVED** [review]

**Review by NguyenVanViet2004 — APPROVED** [review]

**Pull request merged**